ts/components/InstabugButtons.tsx

```diff
@@ -3,34 +3,53 @@ import * as React from "react";
 import { connect } from "react-redux";
 
 import { Button } from "native-base";
+import { Dispatch } from "../store/actions/types";
 import { GlobalState } from "../store/reducers/types";
+import {
+  closeInstabugReport,
+  openInstabugReport
+} from "./../store/actions/debug";
 import IconFont from "./ui/IconFont";
 
 interface OwnProps {
   color?: string;
 }
 
-type Props = ReturnType<typeof mapStateToProps> & OwnProps;
+type Props = ReturnType<typeof mapStateToProps> &
+  OwnProps &
+  ReturnType<typeof mapDispatchToProps>;
 
 class InstabugButtonsComponent extends React.PureComponent<Props, {}> {
   private handleIBChatPress() {
+    this.props.dispatchIBReportOpen("chat");
     Chats.show();
   }
 
   private handleIBBugPress() {
+    this.props.dispatchIBReportOpen("bug");
     BugReporting.showWithOptions(BugReporting.reportType.bug, [
       BugReporting.option.commentFieldRequired
     ]);
   }
 
+  public componentDidMount() {
+    // Register to the instabug dismiss event (https://docs.instabug.com/docs/react-native-bug-reporting-event-handlers#section-after-dismissing-instabug)
+    // This event is fired when chat or bug screen is dismissed
+    BugReporting.onSDKDismissedHandler(
+      (dismissType: string, reportType: string): void => {
+        this.props.dispatchIBReportClosed(reportType, dismissType);
+      }
+    );
+  }
+
   public render() {
     return (
       this.props.isDebugModeEnabled && (
         <React.Fragment>
-          <Button onPress={this.handleIBChatPress} transparent={true}>
+          <Button onPress={() => this.handleIBChatPress()} transparent={true}>
```

Comment:
```suggestion
          <Button onPress={this.handleIBChatPress} transparent={true}>
```

Reply:
@fpersico may arrow function needed because in that function (_handleIBChatPress_) the code refers to **this**?

Reply:
You have to define the function like

```
private handleIBChatPress = () => {
    this.props.dispatchIBReportOpen("chat");
    Chats.show();	    Chats.show();
  }	  }
``